24H Study Room/Protocols/FireAuthServicing.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  FireAuthServicing.swift
+//  24H Study Room
+//
+//  Created by Jia-Jiuan Tsai on 2022-09-05.
+//
+
+import Foundation
+
+protocol FireAuthServicing {
```

Comment:
As the app is written currently, there are not enough levels of abstraction between Firebase and the rest of the app. A good protocol should not expose the implementation details of the types that implement it. The views, view models, etc, should not have to care about the fact that Firebase is the underlying library and platform used for login. You should be able to replace Firebase with another platform without having to change anything outside of the service code. 

So the protocol should not be called `FireAuthServicing`, but something more generic like `AuthServicing`. Then perhaps the specific Firebase implementation of that protocol could be called `FireAuthService`, and maybe you want to have alternate implementations - `SomeOtherAuthService` - and of course `MockAuthService`. 

Then, you should avoid exposing types from the Firebase library outside of the Firebase-specific implementation of this service. So these methods should preferably not return the Firebase `User` object, but instead you could create your own `User`. It would then be the responsibility of the Firebase auth service to map between the objects.